### PR TITLE
Mileston 3

### PR DESCRIPTION
**Pull request by AbigiyaTY**
- Created a class `Classroom` with the following:
  - `@label` instance variable and it is initialized in the constructor.
  - Setter and getter for `@label` ( `attr_accessor`).
- Created the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following is implemented:
  - Created the `has-many` side (a classroom has many students).
  - Created the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classroom's students.
- Created a class `Book` with the following:
  - `@title` and `@author` instance variables and initialized in the constructor.
  - Setters and getters for instance variables (`attr_accessor`).
- Created a class `Rental` with the following:
  - `@date` instance variable, and initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Created the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following is implemented:
  - Created the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Created the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Created the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Created the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modified the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def student(students)
+    @students = students
+    students.classroom = self
+  end
```
Comment (ShadyShawkat) [inline classroom.rb]: - [ ] Kindly make sure that this association method adds a student to the `students` array. You can use something like the following:
```ruby
def add_student(student)
    @students << student
    student.classroom = self
end
```

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :data, :book, :person
+
+  def initialize(data)
+    @data = data
+    @book = book
+    book.rental << self
+    @person = person
+    person.rental << self
+  end
```
Comment (ShadyShawkat) [inline rental.rb]: - [ ] The `Rental` constructor should accept three parameters which are `date`, `book`, and `person`. Currently, you're initializing the rental with a variable called `book` which doesn't exist. Kindly fix this. The `Rental` constructor should look like the following:
```ruby 
def initialize(date, book, person)
   ...
end
```

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @AbigiyaTY, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
### Hello @AbigiyaTY 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is now professional :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------

**Pull request merged**